src/cart/src/cart/crt_iv.c

```diff
@@ -3169,6 +3178,9 @@ crt_iv_update_internal(crt_iv_namespace_t ivns, uint32_t class_id,
 put:
 	iv_ops->ivo_on_put(ivns, NULL, priv);
 exit:
+	if (sync_type.ivs_comp_cb)
+		sync_type.ivs_comp_cb(sync_type.ivs_comp_cb_arg);
+
```

Comment:
i think this portion we should only call if rc != 0

Reply:
Hmm, I already reset the ivs_comp_cb to NULL once the ivs_comp_cb is taken care of inside crt_ivsync_rpc_issue(), so here if ivs_comp_cb is not NULL, it is safe to call?

Reply:
This is why i dont like us changing incoming sync_type and changing comp_cb to null, as it creates these side-effects where we have to keep remembering what we did elsewhere.

It also creates a problem that when we reset values of these to null, we do it before we even can guarantee that ivs_comp_cb() will end up being executed and we wont run into any other errors before we get to it to execute. 

The way i see logic either the com_cb is called from sync handler when its done, or it is called if the caller encountered any error in the process, in which case rc here would be non 0